examples/app_commands/slash_partial_autocomplete.py

```diff
@@ -0,0 +1,55 @@
+from functools import partial
+from os import getenv
+
+from dotenv import load_dotenv
+
+import discord
+from discord.ext import commands
+
+load_dotenv()
+
+bot = discord.Bot()
+
+fruits = ["Apple", "Banana", "Orange"]
+vegetables = ["Carrot", "Lettuce", "Potato"]
+
+
+async def food_autocomplete(
+    ctx: discord.AutocompleteContext, food_type: str
+) -> list[discord.OptionChoice]:
+    items = fruits if food_type == "fruit" else vegetables
+    return [
+        discord.OptionChoice(name=item)
+        for item in items
+        if ctx.value.lower() in item.lower()
+    ]
+
+
+class FoodCog(commands.Cog):
+    @commands.slash_command(name="fruit")
+    async def get_fruit(
+        self,
+        ctx: discord.ApplicationContext,
+        choice: discord.Option(
```

Comment:
Maybe using the ``@option`` decorator would be a better example.

Reply:
Yeah, I was going to change it, because this breaks CI since t doesn't use Annotated, so couldn't get merged